### ISSUE TITLE
Update to support multiple league "types"

### ISSUE THREAD

**Issue by ymhr** [ymhr/mas-league#69]
tl;dr support 'agility', 'showing' and 'other' leagues.

- Put a `types` array on the league record in the `leagues` collection. This array represents the supported types.
- When a user is adding a dog, they can tick the types that they compete in. Only agility will have a grade option.
- When the admin puts a dog in a league, it will insert them into the sub-types they have ticked. (see note 1)
- When a user adds a 'run' (should that word change?) that needs the same checkboxes so that the user can specify what they competed in. The cloud function should then update the correct type in the dogs data.

Notes:
1. When a dog is added to a league, the map representing that needs to be broken down into their types, e.g.

```json
{
  "leagues": {
    "2019": {
      "agility": {
        "grade": 2,
        "points": 20
      },
      "showing": {
        "points": 12
      }
    }
  }
}
```
